app/javascript/components/TrackList.jsx

```diff
@@ -0,0 +1,48 @@
+import React from 'react'
+import { Track } from './TrackList/Track'
+import { TextFilter } from './TextFilter'
+import { useList } from '../hooks/use_list'
+import { useRequestQuery } from '../hooks/request_query'
+
+export function TrackList(props) {
+  const [request, setFilter, setSort, setPage] = useList(props.request)
+  const { status, data, isFetching } = useRequestQuery('track-list', request)
+
+  return (
+    <div className="track-list">
+      <TextFilter
+        filter={request.query.filter}
+        setFilter={setFilter}
+        id="track-list-filter"
+        placeholder="Search language tracks"
+      />
+      {status === 'error' && <p>Something went wrong</p>}
+      {isFetching && <p>Loading</p>}
```

Comment:
I like the use of `isFetching`, which is more declarative than `status == "loading"`. Looking at [the documentation](https://react-query.tanstack.com/docs/api#usequery), it looks like there are also `isError` and `isSuccess` properties being returned. Maybe we could use them to replace the status? 

If you think this is a nice improvement, let me know and I'll create an issue to update them elsewhere too.

Reply:
Cool! I think it makes sense to do that :)